client/my-sites/email/titan-mail-quantity-selection/index.jsx

```diff
@@ -287,151 +224,140 @@ class TitanMailQuantitySelection extends React.Component {
+						'All of your mailboxes are due to renew at the regular price of {{strong}}%(fullPrice)s{{/strong}} per mailbox when your subscription renews on {{strong}}%(expiryDate)s{{/strong}}.',
 						{
 							args: {
-								date: moment( expiryDate ).format( 'LL' ),
-								price: renewalCost.text,
+								fullPrice: renewalCost.text,
+								expiryDate: moment( expiryDate ).format( 'LL' ),
 							},
 							components: {
 								strong: <strong />,
 							},
 							comment:
-								'%(price)s is a formatted price for each mailbox, e.g. $3.50, €3.75, or PLN 4.50; ' +
-								'%(date)s is a localized date of the rough form: 17 February 2021',
+								'%(fullPrice)s is a formatted price for a monthly email mailbox subscription (e.g. $3.50, €3.75, or PLN 4.50), %(expiryDate)s is a localized date (e.g. February 17, 2021)',
 						}
 					) }
-				</span>
-				<CompactCard className="titan-mail-quantity-selection__prorating-details">
-					<Gridicon icon="info-outline" size={ 18 } />
-					<span>{ prorationMessage }</span>
-				</CompactCard>
-			</React.Fragment>
+				</>
+			</Notice>
 		);
 	}
 
 	renderForm() {
-		const { isLoadingDomains, selectedDomainName, titanMonthlyProduct, translate } = this.props;
+		const { isLoadingDomains, titanMonthlyProduct, translate } = this.props;
 
 		if ( isLoadingDomains || ! titanMonthlyProduct ) {
 			return <AddEmailAddressesCardPlaceholder />;
 		}
 
 		return (
 			<>
-				<SectionHeader label={ translate( 'Choose Mailbox Quantity' ) } />
+				<SectionHeader label={ translate( 'Add New Mailboxes' ) } />
 
 				<Card>
-					<div className="titan-mail-quantity-selection__domain-info">
-						{ translate( "You're adding mailboxes for {{strong}}%(domainName)s{{/strong}}", {
-							args: {
-								domainName: selectedDomainName,
-							},
-							components: {
-								strong: <strong />,
-							},
-							comment: '%(domainName)s is a domain name',
-						} ) }
-					</div>
-					<div className="titan-mail-quantity-selection__mailbox-info">
-						{ this.renderCurrentMailboxCounts() }
-					</div>
-					<div>
-						<FormLabel>{ translate( 'Number of new mailboxes to add' ) }</FormLabel>
+					<div className="titan-mail-quantity-selection__form">
+						<FormLabel htmlFor="quantity">{ translate( 'Number Of Mailboxes' ) }</FormLabel>
+
 						<FormTextInput
 							name="quantity"
+							id="quantity"
 							type="number"
 							min="1"
 							max="50"
 							step="1"
-							placeholder={ translate( 'Number of new mailboxes' ) }
+							placeholder={ translate( 'Number of mailboxes' ) }
 							value={ this.state.quantity }
 							onChange={ this.onQuantityChange }
 						/>
-					</div>
-					<hr className="titan-mail-quantity-selection__divider titan-mail-quantity-selection__divider-pricing" />
-					<div className="titan-mail-quantity-selection__pricing-info">
-						{ this.renderPricingDetails() }
+
+						<Button onClick={ this.onQuantityChange }>
```

Comment:
Does it make sense to have this `Button` and/or the interior `span` be given an `id` or some similar way of confirming which element got clicked in `onQuantityChange()`? The current `instanceof` checks in that code feel fragile to me.

Actually, why not add a dedicated handler for this action? The weird thing here is that `onQuantityChange()` is reacting to the user doing something to the actual `input`, while in this case, we're doing a very controlled modification to the state. If necessary, the parsing logic to fetch the current value can be moved to a helper function.

Reply:
You're right, using a dedicated handler is a much better approach! I'll update that code.